authorization/invitation/model/invitation_model_service.go

```diff
@@ -0,0 +1,201 @@
+		}
+	}
+
+	// Iterate through all of the invitations and confirm that for each one:
+	// 1) a valid user has been specified via its User ID, e-mail address or username
+	// 2) any roles specified are valid roles for the organization, team or security group
+	// For each invitation, ensure that the IdentityID value can be found and set it
+	for i, invitation := range invitations {
+		// If the UserID has been provided, confirm it is valid and that the identity is a user
+		if invitation.IdentityID != nil {
+			identity, err := repo.Identities().Load(ctx, *invitation.IdentityID)
+			if err != nil {
+				return errors.NewInternalErrorFromString(ctx, fmt.Sprintf("invalid identity ID specified: %s", invitation.IdentityID))
+			}
+
+			if !identity.IsUser() {
+				return errors.NewBadParameterErrorFromString("Identity ID", invitation.IdentityID, "identity is not a user")
+			}
+		} else if invitation.UserName != nil {
+			// If the username has been provided, confirm the user is valid and that the identity is a user, and set the UserID
+			identities, err := repo.Identities().Query(account.IdentityFilterByUsername(*invitation.UserName))
+			if err != nil {
+				return errors.NewInternalError(ctx, err)
+			}
+			// If there is anything other than 1 result found, we have a problem
+			if len(identities) == 0 {
+				// If no users are found, return an error
+				return errors.NewBadParameterErrorFromString("Username", invitation.UserName, "username not found")
+			}
+
+			// Set the IdentityID to that of the identity found
+			invitations[i].IdentityID = &identities[0].ID
+		} else if invitation.UserEmail != nil {
+			// If the user's e-mail address has been provided, confirm the user is valid and that the identity is a user, and set the UserID
+			users, err := repo.Users().Query(account.UserFilterByEmail(*invitation.UserEmail))
+			if err != nil {
+				return errors.NewInternalError(ctx, err)
+			}
+			// We expect exactly 1 user to be found, if not we return an error
+			if len(users) == 0 {
+				return errors.NewBadParameterErrorFromString("E-mail", invitation.UserEmail, "user with e-mail address not found")
+			} else if len(users) > 1 {
+				return errors.NewBadParameterErrorFromString("E-mail", invitation.UserEmail, "more than one user with e-mail address found")
+			}
+
+			userID := &users[0].ID
+
+			// Now that we have the user ID, we can lookup the identity ID
+			identities, err := repo.Identities().Query(account.IdentityFilterByUserID(*userID))
+			// If there is anything other than 1 result found, return an error
+			if len(identities) == 0 {
+				return errors.NewBadParameterErrorFromString("E-mail", invitation.UserEmail, "no identity found for user with e-mail address")
+			} else if len(identities) > 1 {
+				return errors.NewBadParameterErrorFromString("E-mail", invitation.UserEmail, "more than one identity found for user with e-mail address")
```

Comment:
It's not an error. We support multiple Identities per user. We don't use it right now though. This is for future use of remote identities (from GitHub, JIRA and other issue trackers).
You need to look up for identities with account.KeycloakIDP identity provider. Identities with that provider come from Keycloak/RHD and they represent OSIO users. There is supposed to be only one identity with such provider per user.
Feel free to add missing methods to the identity repo for that ;) But again, I don't think we should have look up for useremail at all in this API in the first place.